app/controllers/concerns/orders_service.rb

```diff
@@ -0,0 +1,55 @@
+module OrdersService
+  def add_order_items
+    @cart = rebuild_items_list
+    return :unvalid_qty unless valid_item_list
+
+    order = Order.create(order_params)
+
+    return order_created if order
+  end
+
+  private
+  def order_params
+    params.require(:order)
+          .permit(:total_price, :user_id,
+                  order_items_attributes: [:product_id,
+                                           :quantity,
+                                           :current_price])
+  end
+
+  def valid_item_list
+    @cart.each do |item|
+      if qty_exceeds_range item["product_id"], item["quantity"]
+        flash[:danger] = t "order.unvalid_qty"
+        return false
+      end
+    end
+
+    true
+  end
+
+  def rebuild_items_list
+    result = JSON.parse(params[:order][:order_items_attributes])
+    params[:order][:order_items_attributes] = result
+
+    result
```

Comment:
```suggestion
   JSON.parse(params[:order][:order_items_attributes])
```

chị nghĩ chỉ cần ri nhỉ? 

hoặc 
```suggestion
    params[:order][:order_items_attributes] = JSON.parse(params[:order][:order_items_attributes])
```
nếu dài quá thì xún line trống trước dấu `=`